internal/testcase/case_test.go

```diff
@@ -353,3 +358,250 @@ func TestLoadTestSteps(t *testing.T) {
+		"ephemeral exists and no permissions at all": {
+			cl:                    clientWithExistingNsNoPerms,
+			expectedCleanupErrors: 1,
+			getNsBeforeCleanup: func(t *testing.T, err error) {
+				require.NoError(t, err)
+			},
+			getNsAfterCleanup: func(t *testing.T, err error) {
+				require.NoError(t, err)
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			c := NewCase(name, "", tt.options...)
+			tm := &testMock{}
+			cl := tt.cl(t, c.ns.name)
+			if npc, ok := cl.(*noPermClient); ok {
+				npc.t = t
+			}
+			clk := clientWithKubeConfig{
+				Client:         cl,
+				kubeConfigPath: "kubeconfig/path",
+				logger:         testutils.NewTestLogger(t, ""),
+			}
+
+			gotErr := c.createNamespace(tm, clk)
+			if tt.wantErr == nil {
+				assert.NoError(t, gotErr)
+			} else {
+				assert.ErrorIs(t, gotErr, tt.wantErr)
+			}
+
+			baseClient := cl
+			if npc, ok := cl.(*noPermClient); ok {
+				baseClient = npc.Client
+			}
+
+			ns := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: c.ns.name,
+				},
+			}
+			err := baseClient.Get(t.Context(), kubernetes.ObjectKey(ns), ns)
+			tt.getNsBeforeCleanup(t, err)
+
+			if tm.cleanup != nil {
+				tm.cleanup()
+			}
+
+			assert.Len(t, tm.testError, tt.expectedCleanupErrors)
+			ns = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: c.ns.name,
+				},
+			}
+			err = baseClient.Get(t.Context(), kubernetes.ObjectKey(ns), ns)
+
+			tt.getNsAfterCleanup(t, err)
+		})
+	}
+}
+
+// noPermClient wraps a client and returns forbidden errors for Create/Delete operations.
+// Optionally it also refuses Get operations.
+type noPermClient struct {
+	client.Client
+	forbidGet bool
+	t         *testing.T
+}
+
+var errCreationForbidden = k8serrors.NewForbidden(schema.GroupResource{Group: "", Resource: "namespaces"}, "foo", fmt.Errorf("forbidden: User cannot create resource \"namespaces\""))
+
+func (c *noPermClient) Create(_ context.Context, obj client.Object, _ ...client.CreateOption) error {
+	c.t.Logf("Create object %v refused", obj.GetObjectKind().GroupVersionKind())
+	return errCreationForbidden
+}
+
+func (c *noPermClient) Delete(_ context.Context, obj client.Object, _ ...client.DeleteOption) error {
+	c.t.Logf("Delete object %v refused", obj.GetObjectKind().GroupVersionKind())
+	return k8serrors.NewForbidden(schema.GroupResource{Group: "", Resource: "namespaces"}, obj.GetName(), fmt.Errorf("forbidden: User cannot delete resource \"namespaces\""))
+}
+
+func (c *noPermClient) Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+	if c.forbidGet {
+		c.t.Logf("Get object %v refused", key)
+		return k8serrors.NewForbidden(schema.GroupResource{Group: "", Resource: "namespaces"}, obj.GetName(), fmt.Errorf("forbidden: User cannot get resource \"namespaces\""))
+	}
+	return c.Client.Get(ctx, key, obj, opts...)
+}
+
+func clientWithExistingNsNoWritePerm(t *testing.T, nsName string) client.Client {
```

Comment:
nit?

```suggestion
func newClientWithExistingNsNoWritePerm(t *testing.T, nsName string) client.Client {
```

Same for the other functions.